crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4567,6 +4567,22 @@ impl<'db> TypeInferenceBuilder<'db> {
                 let param_type = self.infer_type_expression(parameters);
                 UnionType::from_elements(self.db, [param_type, Type::none(self.db)])
             }
+            KnownInstanceType::Union => {
+                let mut builder = UnionBuilder::new(self.db);
+                match parameters {
+                    ast::Expr::Tuple(t) => {
+                        builder = t
+                            .elts
+                            .iter()
+                            .map(|elt| self.infer_type_expression(elt))
+                            .fold(builder, super::builder::UnionBuilder::add);
+                        builder.build()
+                    }
+                    _ => {
+                        UnionType::from_elements(self.db, [self.infer_type_expression(parameters)])
+                    }
```

Comment:
I don't think there are any tests for this arm? It would mean adding a test with an annotation like `x: Union[int]`.

Pyright emits a diagnostic on this, mypy doesn't. I don't really feel like it should be an error -- it is a useless use of `Union`, but there is a unique and clear interpretation. There could be a lint rule against it.

I agree with @MichaReiser that I don't see why we use `UnionType::from_elements` here when it should always simplify back to a single type. I think this is completely equivalent, but more efficient:
```suggestion
                    _ => self.infer_type_expression(parameters)
```